apps/devhub/tests/test_views.py

```diff
@@ -2222,6 +2222,30 @@ def test_unparsable_xpi(self, flag_is_active, v):
              for m in data['validation']['messages']],
             [(u'Could not parse install.rdf.', True)])
 
+    @mock.patch('devhub.tasks.run_validator')
+    def test_experiment_xpi_allowed(self, mock_validator):
+        user = UserProfile.objects.get(email='regular@mozilla.com')
+        self.grant_permission(user, 'Experiments:submit')
+        mock_validator.return_value = json.dumps(self.validation_ok())
+        self.upload_file('../../../files/fixtures/files/experiment.xpi')
+        upload = FileUpload.objects.get()
+        response = self.client.get(reverse('devhub.upload_detail',
+                                           args=[upload.uuid, 'json']))
+        data = json.loads(response.content)
+        assert data['validation']['messages'] == []
+
+    @mock.patch('devhub.tasks.run_validator')
+    def test_experiment_xpi_not_allowed(self, mock_validator):
+        mock_validator.return_value = json.dumps(self.validation_ok())
+        self.upload_file('../../../files/fixtures/files/experiment.xpi')
+        upload = FileUpload.objects.get()
+        response = self.client.get(reverse('devhub.upload_detail',
+                                           args=[upload.uuid, 'json']))
+        data = json.loads(response.content)
+        assert data['validation']['messages'] == [
+            {u'tier': 1, u'message': u'You cannot submit this type of add-ons',
+             u'fatal': True, u'type': u'error'}]
```

Comment:
Likewise it might be good to check that it doesn't exist in the database but in this case I'm fairly confident it wouldn't be.